From 839ad29b16a8baf7b1470f13d7faa0ce941769b3 Mon Sep 17 00:00:00 2001
From: Jan Vorlicek <janvorli@microsoft.com>
Date: Mon, 30 Nov 2020 16:11:36 +0100
Subject: [PATCH] Fix build on Alpine edge (#45352)

The atoll definition in the pal.h was leaking into the PAL implementation and
on Alpine edge, the difference in throws() classification was causing a build
break.
---
 src/coreclr/src/pal/src/include/pal/palinternal.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/coreclr/src/pal/src/include/pal/palinternal.h b/src/coreclr/src/pal/src/include/pal/palinternal.h
index e73720bec73..4ad42a03a13 100644
--- a/src/coreclr/src/pal/src/include/pal/palinternal.h
+++ b/src/coreclr/src/pal/src/include/pal/palinternal.h
@@ -170,6 +170,7 @@ function_name() to call the system's implementation
 #define memset DUMMY_memset
 #define memmove DUMMY_memmove
 #define memchr DUMMY_memchr
+#define atoll DUMMY_atoll
 #define strlen DUMMY_strlen
 #define stricmp DUMMY_stricmp
 #define strstr DUMMY_strstr
@@ -357,6 +358,7 @@ function_name() to call the system's implementation
 #undef memset
 #undef memmove
 #undef memchr
+#undef atoll
 #undef strlen
 #undef strnlen
 #undef wcsnlen
-- 
2.30.2

